components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -948,4 +948,17 @@ void AIChatService::OpenConversationWithStagedEntries(
   conversation->MaybeFetchOrClearContentStagedConversation();
 }
 
+void AIChatService::AssociateContent(
+    ConversationHandler::AssociatedContentDelegate* content,
+    const std::string& conversation_uuid) {
+  CHECK(content);
+
+  auto* conversation = GetConversation(conversation_uuid);
```

Comment:
Using the non-async version of `GetContent` will mean this only works if the conversation has active UI. That seems fine for this feature, but maybe leave a comment?